### PR TITLE
Use controller methods for handling the encyrption keys from agent

### PR DESCRIPTION
**Pull request by sanimej**
Instead of using Provider interface methods to get the encryption keys added a controller function that will be called by the daemon.

Signed-off-by: Santhosh Manohar <santhosh@docker.com>

### CODE REVIEW

**Comment by mrjana** [comment]
LGTM

File: controller.go [hunk 1]
```diff
@@ -220,55 +225,33 @@ func isValidClusteringIP(addr string) bool {
...
+	if len(c.keys) == 0 {
+		c.keys = keys
+		if c.agent != nil {
+			return (fmt.Errorf("libnetwork agent setup without keys"))
+		}
+		if c.clusterConfigAvailable {
+			return c.agentSetup()
+		}
+		log.Debugf("received encryption keys before cluster config")
+	}
```
Comment (aboch) [inline controller.go]: I think this should be an else

Comment (sanimej) [inline controller.go]: Good catch. Thanks. Earlier when I didn't have the check `c.clusterConfigAvailable {` golint complained about having an else. So I removed it. But its needed now.

File: controller.go [hunk 2]
```diff
@@ -220,55 +225,34 @@ func isValidClusteringIP(addr string) bool {
...
+		if c.agent != nil {
+			return (fmt.Errorf("libnetwork agent setup without keys"))
+		}
+		if c.clusterConfigAvailable {
+			return c.agentSetup()
+		}
+		log.Debugf("received encryption keys before cluster config")
+		return nil
+	}
+	return c.handleKeyChange(keys)
```
Comment (mavenugo) [inline controller.go]: Please make sure c.agent is not nil before handling key-rotation

**Comment by mavenugo** [comment]
LGTM

**Pull request merged**